app/repos/dogs.js

```diff
@@ -33,6 +34,7 @@ const createDogs = async (dogs, owners, enforcement, transaction) => {
       const breed = await getBreed(dog.breed)
 
       const dogEntity = await sequelize.models.dog.create({
+        id: dog.indexNumber ?? undefined,
```

Comment:
Is it not dangerous to auto-create the index number if it hasn't been passed through from the spreadsheet? Either the '?? undefined' isn't necessary (because the earlier validation catches a missing index number), or you shouldn't need the '?? undefined'